src/utils/ioHandler.js

```diff
@@ -0,0 +1,11 @@
+import { Console } from "@woowacourse/mission-utils"
+
+async function input(message){
+    return await Console.readLineAsync(message);
```

Comment:
```suggestion
function input(message){
    return Console.readLineAsync(message);
```
`input` 함수에 로직이 1개 밖에 없어서 `async/await`을 생략하는건 어떻게 생각하시나요??

Reply:
입출력이 비동기적이어서 생략할 시에 의도된 값을 변수에 할당하지 못하지 않나요..?

Reply:
지금은 한번 추상화 되어있는 `input` 함수이고 내부 로직이 입력을 받는 로직밖에 없기 때문에 외부에서 사용할 때 `await`키워드만 사용해준다면 정상적으로 Promise를 다룰 수 있을 것 같습니다!

```js
async function input(message){
    const input = await Console.readLineAsync(message);
    // 추가적인 로직
    return input
}
```
이런 경우에는 `input` 변수에 `Console` 입력 값이 들어와야 하기 때문에 `await`으로 기다려주는게 맞는 것 같아요!

Reply:
리뷰해주신 내용 적용해봤는데, 말씀하신대로 잘 동작하네요..!
리뷰 감사합니다 :)
```
async function input(message){
    const userInput = await Console.readLineAsync(message);
    // 추가적인 로직
    return userInput;
}
```

말씀해주신 이 경우도 Console.readLineAsync의 반환값이 Promise이기 때문에
결론적으로 input 반환값에 Promise 객체가 들어가게 돼서 정상적으로 동작이 됩니다!
그래서 await으로 기다려줄 필요가 없긴 합니다.

그런데 위 코드를 고민하다보니, Promise 객체를 다루고 있다는 걸 명시적으로 설명하기 위해서
async, await을 써주는 것도 좋겠다는 생각이 들었어요! 코드의 사이즈가 커졌을 때, async await을 보고
저 함수가 비동기, 즉 Promise 객체를 다루고 있다는게 명확해질 것 같아서요

Reply:
> 말씀해주신 이 경우도 Console.readLineAsync의 반환값이 Promise이기 때문에 결론적으로 input 반환값에 Promise 객체가 들어가게 돼서 정상적으로 동작이 됩니다! 그래서 await으로 기다려줄 필요가 없긴 합니다.
> 

```js
function input(message){
    const userInput = Console.readLineAsync(message);
    const array = userInput.split(""); // userInput은 pending 상태로 아직 값이 안들어옴
    return array;
}
```
만약 여기서 `await` 키워드를 사용하지 않는다면 `const array = userInput.split("");` 여기 로직에서 `userInput`은 아직 `pending` 상태이기 때문에 올바르게 Promise를 다루지 못하기 때문에 `await`을 사용해서 `userInput` 변수에 값이 들어오기를 기다려야 한다고 생각하는데 종빈님의 생각은 다르신가요??

그리고 말씀해주신 말씀해주신 비동기 로직을 다루는 걸을 명확히 하고 싶다면 함수 이름을 `inputAsync`와 같이 변경해서 비동기 로직을 다룬다는걸 외부에 알리면 어떨까요??

지금 로직에서 `async/await`은 사실상 안써도 되는 곳에 허공에 사용하는 느낌이 있다보니깐  함수 이름 변경을 고려해보는게 가장 좋을 것 같다고 생각하는데 종빈님의 생각은 어떠신가요??

Reply:
위에 만들어주신 코드 내에서는 await이 필수적으로 있어야 할 것 같네요..!
```
function input(message){
    const userInput = Console.readLineAsync(message);
    return userInput;
}
```
await으로 기다려줄 필요가 없다고 한 건 제가 위에 적은 코드입니다..!

---

`inputAsync`와 같은 변수 작명은 사람의 코드 스타일에 따라서 갈 것 같긴 하네요.
이전에 비슷한 고민을 했던 적이 있는데, 예를 들어서 커스텀 훅을 짤 때

커스텀 훅
```
const customHook = () => {
   speak = () => {
        ...
    }
    listen = () => {
        ...
    }
    return {speak, listen}
}
```
첫번째 방법
```
const person = customHook();
person.speak();
person.listen();
```

두번째 방법
```
const {personSpeak: speak, personListen} = customHook();
personSpeak();
personListen();
```
첫번째 방법과 두번째 방법이 각각 객체의 특징을 사용한 명시, 변수명을 통한 명시에서 차이를 주고 있다고 생각합니다.
이런 상황에서 같이 프로젝트를 진행하던 동료들끼리 결정한 방식은 "본인 스타일에 더 잘맞는 걸 선택하자"였습니다.

지금 Promise를 처리하는 방법론이 위에 제가 했던 고민이랑 조금은 결이 비슷한 것 같아서 코멘트 달아봤습니다!

Reply:
> 위에 만들어주신 코드 내에서는 await이 필수적으로 있어야 할 것 같네요..!
> 
> ```
> function input(message){
>     const userInput = Console.readLineAsync(message);
>     return userInput;
> }
> ```

아하 이런 상황을 말씀 해주신거였군요! 저는 저 사이에 추가 로직이 들어가는 상황에서는 `await`이 필요할 것 같다는 의미였는데 소통의 오류가 있었던 것 같습니다🙏🏻

> async await을 보고 저 함수가 비동기, 즉 Promise 객체를 다루고 있다는게 명확해질 것 같아서요

이 부분에서 제가 함수 이름 변경이 가장 좋다고 생각했던 이유는 `외부에서 사용할 때 알 수 있을까?` 라는 고려했던 것 같아요!

1. 함수 이름 변경
```js
const input = await inputAsync();
```
- 함수 이름을 통해 내부 로직을 확인하지 않고도 Promise를 다루는 비동기 함수임을 알 수 있음

2. 추상화된 메서드 내부에서 `async/await` 사용
```js
const input = await input(); 
```
- 외부에서는 `input` 함수가 비동기 로직을 다루는지 알 수 없음, 내부 로직을 봐야 알 수 있음

이렇게 사용하는 외부에서 Promise를 다루는 비동기 함수임을 알 수 있다는 점에서 `함수 이름 변경`이 적합하다고 생각하게 되었는데 종빈님은 어떠한 이유로 2번 방식이 Promise 객체를 다루고 있다는게 명확하다고 생각하셨는지 들어볼 수 있을까요?

Reply:
좋은 예시인 것 같아요..!
결론적으로 보면 "`Promise`를 바로 반환하기 때문에 `async, await`을 쓰지 않아도 되는 함수에
명시적으로 `async, await`을 적는 방법이 좋을까?"라는 질문이 되는 것 같아요
제 개인적인 견해는 "체이닝 느낌으로 async, await을 명시적으로 적고 함수 스택 가장 아래쪽까지 파고들면서,
이 흐름이 비동기적인 과정을 거치는 흐름이라는 걸 알려주는 것도 좋을 것 같다"인 것 같아요

Reply:
충분히 납득할 수 있는 근거인 것 같습니다 좋은 의견 감사합니다👍👍👍